client/src/modules/install/install.js

```diff
@@ -0,0 +1,51 @@
+angular.module('bhima.controllers')
+  .controller('InstallApplicationController', InstallApplicationController);
+
+// dependencies injection
+InstallApplicationController.$inject = ['InstallService', '$state', 'NotifyService'];
+
+// controller definition
+function InstallApplicationController(InstallService, $state, Notify) {
+  var vm = this;
+
+  vm.setup = { enterprise : {}, project : {}, user : {} };
+
+  // expose methods to the view
+  vm.submit = submit;
+
+  InstallService.checkBasicInstallExist()
+    .then(processCheckResult)
+    .catch(Notify.handleError);
+
+  function processCheckResult(alreadyExist) {
+    vm.alreadyExist = alreadyExist;
+
+    if (alreadyExist) { $state.go('login'); }
+  }
+
+  function notifyInstallSucess() {
+    Notify.success('INSTALL.SUCCESS_INSTALL');
+    $state.go('login');
+  }
+
+  function checkPassword() {
+    if (!vm.setup.user.password) { return false; }
+
+    if (vm.setup.user.password !== vm.setup.user.repassword) {
```

Comment:
Do we not have a service for this?  Or a directive/component?

Reply:
It looks like [UserService.validatePassword()](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/users/users.service.js#L124) will do this for you.  Could you use that?